src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -260,10 +235,11 @@ public ContentItemGroup FindBestItemGroup(SelectionCriteria criteria, params Pat
             return null;
         }
 
-        private IList<ContentItem> FindItemsImplementation(PatternSet definition, IEnumerable<Asset> assets)
+        private List<ContentItem> FindItemsImplementation(PatternSet definition, IEnumerable<Asset> assets)
```

Comment:
I'm curious how does this improve things? 

The ContentItemGroup constructor takes IList anyways.

Reply:
`List<T>.GetEnumerator()` returns a struct. Anything (like `IList<T>`) that uses `IEnumerator<T>` will require a memory allocation.

Unfortunately it's also true for `IReadOnly[List|Dictionary]`, so then there's a tradeoff between cost of enumeration vs compiler enforced correctness. Or stop using `foreach` and LINQ (or anything else that uses `IEnumerable<T>`), and just use an old-school `for` loop instead.

For people in Microsoft: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/24181/Garbage-Collection?anchor=%E2%9D%8C-**avoid**-boxing-enumerators-of-built-in-types

Reply:
I get that, but ContentItemGroup takes an IList, so we're not saving on an allocation.

Reply:
You're right. It's left over from an earlier version of the PR where the code was actually using the value in a `foreach`, where it did make a difference. Seems I then went and changed it to store the returned collection directly rather than copying items into a new collection, and never reverted this back. Personally I prefer using concrete types on private APIs so that internal consumers have maximum type information, and using interfaces externally so we have the ability to change implementation behind that abstraction without breaking consumers. But we can revert this line if you prefer.

Reply:
I don't mind it since it's a private method anyways. 

I mostly curiuous whether I'm missing out on a new trick :D